compiler/optimizer/OrderBlocks.cpp

```diff
@@ -164,8 +164,7 @@ bool TR_OrderBlocks::safeToMoveAway(TR::Block *block, TR::Block *prevBlock)
    if (_changeBlockOrderBasedOnHWProfile)
       return true;
 #if 1
-    if (hasOtherHotAsmFlowPredeccessors(block, prevBlock) ||
```

Comment:
Yes, in my opinion. If there are knock on cleanup opportunities like the one you mentioned, then we should do those things too.